catchup/service.go

```diff
@@ -201,6 +237,10 @@ func (s *Service) innerFetch(r basics.Round, peer network.Peer) (blk *bookkeepin
 //  - If the block is already in the ledger (e.g. if agreement service has already written it)
 //  - If the retrieval of the previous block was unsuccessful
 func (s *Service) fetchAndWrite(r basics.Round, prevFetchCompleteChan chan bool, lookbackComplete chan bool, peerSelector *peerSelector) bool {
+	// If sync-ing this round would break our cache invariant, don't fetch it
+	if set, syncRound, err := s.GetSyncRound(); err == nil && set && r >= basics.Round(syncRound+s.cfg.MaxAcctLookback) {
```

Comment:
nit: looks like `syncRound == 0` is a pretty good indication of a fact it wasn't set. Yes, implicit but imo clear enough

Reply:
Presumably it's implied, but to sanity check:  _If_ `syncRoundSet` is removed, then `syncRound` must be reset to 0 in `UnsetSyncRound`.

Reply:
Updated to have 0 be the unset value.